### PR TITLE
Create Calculator.js

### PR DESCRIPTION
**Pull request by ashoo-jindal**
<h1>Calculator Parser Implementation</h1>
<h2>Description</h2>
<p>This pull request introduces a new parser for evaluating mathematical expressions in the @SlackerBot. The parser allows users to input mathematical calculations in a user-friendly format, returning results rounded to two decimal places.</p>

<h2>Features</h2>
<ul>
    <li>Supports basic arithmetic operations: addition, subtraction, multiplication, and division.</li>
    <li>Validates user input to ensure only numeric values and operators are processed.</li>
    <li>Returns the result rounded to two decimal places for improved readability.</li>
    <li>Handles invalid input gracefully by providing user feedback.</li>
</ul>

<h2>Usage</h2>
<p>Users can trigger the parser with the command:</p>
<pre>
    !calculate &lt;expression&gt;
</pre>
<p>For example:</p>
<pre>
    !calculate 5 + 3 * 2
</pre>
<p>This will return:</p>
<pre>
    The result is: 11.00
</pre>

<h2>Contributions</h2>
<p>Feedback and contributions to enhance the parser's functionality are welcome!</p>

### CODE REVIEW

File: Parsers/Calculator.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+activation_example: !calculate <expression>
+regex: !calculate (.+)
```
Comment (earlduque) [inline Parsers/Calculator.js:3]: key:value pairs in the header template are affected by leading spaces.

**Review by earlduque — CHANGES_REQUESTED** [review]

File: Parsers/Calculator.js (hunk 1 above)
Comment (ashoo-jindal) [reply Parsers/Calculator.js]: I have removed the spaces. 
Thanks

**Review by earlduque — APPROVED** [review]

**Comment by earlduque** [comment]
I seem to be getting only the error path when activating it

**Comment by ashoo-jindal** [comment]
I tested this in Scripts - Background like this. What could be the issue? Can you guide a little.

var expression = "5/5*5-5+5.50";

function evaluateExpression(expr) {
    // Remove whitespace
    expr = expr.replace(/\s+/g, '');

    // Basic validation to allow only numbers and operators
    if (!/^[\d+\-*/().]+$/.test(expr)) {
        throw new Error("Invalid expression");
    }

    // Use Function constructor to evaluate the expression safely
    return new Function('return ' + expr)();
}

try {
    var result = evaluateExpression(expression);

    // Round the result to 2 decimal places
    var roundedResult = result.toFixed(2);

    // Send the calculated result back to the user
    gs.info(roundedResult);
} catch (error) {
    // Handle errors if the expression is invalid
    gs.info("Oops! I couldn't understand that. Please provide a valid mathematical expression.");
}

**Comment by ashoo-jindal** [comment]
I think the error can be at line 33
I can change it to this, if that will resolve the issue?
`    new x_snc_slackerbot.Slacker().send_chat(current, 'The result is: ' + roundedResult, false);`

**Comment by ashoo-jindal** [comment]
I have been testing this by Installing this bot on your own workspace and it's not able to evaluate this line
`return new Function('return ' + expr)()`

neither it's able to evaluate this 
```
        var result = eval(expr);
        return result;
```

but both of them works in background scripts.

**Pull request merged**